gis/grass/Portfile

```diff
@@ -1,11 +1,300 @@
+        "
+}
+
+set projdf "if {"
+foreach pv ${proj_versions} {
+    set projdf "${projdf}!\[variant_isset proj${pv}\] && "
+}
+set projdf [string range ${projdf} 0 end-4]
+set projdf "${projdf}} { default_variants +proj${pv} }"
+eval ${projdf}
+
+variant gui description {Build with wxPython GUI and application bundle} {
+    wxWidgets.use           wxWidgets-3.2
+
+    depends_lib-append      port:${wxWidgets.port}
+
+    foreach pyver ${python_suffixes} {
+        if {[variant_isset python${pyver}]} {
+            depends_lib-append  port:py${pyver}-wxpython-4.0
+        }
+    }
+
+    configure.args-append   --with-wxwidgets=${wxWidgets.wxconfig}
+    configure.args-delete   --with-opengl=no
+    configure.args-append   --with-opengl=osx
+
+    # Create app bundle
+    global app_name
+    set app_name_version [join [lrange [split ${version} "."] 0 1] "."]
+    set app_name GRASS-${app_name_version}.app
+
+    post-build {
+        xinstall -d -m 0755 ${worksrcpath}/${app_name}/Contents
+        xinstall -d -m 0755 ${worksrcpath}/${app_name}/Contents/Resources
+        xinstall -d -m 0755 ${worksrcpath}/${app_name}/Contents/MacOS
+
+        set script [open "${worksrcpath}/${app_name}/Contents/MacOS/GRASS" w 0755]
+        puts ${script} "#!/usr/bin/osascript"
+        puts ${script} ""
+        puts ${script} "tell application \"Terminal\" to do script \"${prefix}/bin/grass${main_version} --gui; exit\""
+        close ${script}
+
+        copy ${worksrcpath}/macosx/app/AppIcon.icns \
+            ${worksrcpath}/${app_name}/Contents/Resources/AppIcon.icns
+        copy ${worksrcpath}/macosx/app/GRASSDocument_gxw.icns \
+            ${worksrcpath}/${app_name}/Contents/Resources/GRASSDocument_gxw.icns
+
+        set info_plist ${worksrcpath}/${app_name}/Contents/Info.plist
+        copy ${worksrcpath}/macosx/app/Info.plist.in ${info_plist}
+
+        set version_file        [open ${worksrcpath}/include/VERSION]
+        set version_data        [split [read ${version_file}] "\n"]
+        close ${version_file}
+        set grass_version_major [lindex ${version_data} 0]
+        set grass_version_minor [lindex ${version_data} 1]
+        set grass_version_date  [lindex ${version_data} 3]
+        set bundle_version      [join [lrange ${version_data} 0 2] "."]
+
+        reinplace "s|@GRASS_VERSION_MAJOR@|${grass_version_major}|" ${info_plist}
+        reinplace "s|@GRASS_VERSION_MINOR@|${grass_version_minor}|" ${info_plist}
+        reinplace "s|@GRASS_VERSION_DATE@|${grass_version_date}|" ${info_plist}
+        reinplace "s|@BUNDLE_VERSION@|${bundle_version}|" ${info_plist}
+        reinplace "s|@DEPLOYMENT_TARGET@|${macosx_deployment_target}|" ${info_plist}
+    }
+
+    post-destroot {
+        copy ${worksrcpath}/${app_name} ${destroot}${applications_dir}
+    }
+}
+
+variant mysql5 description {Add MySQL 5 support} {
+    depends_lib-append      path:bin/mysql_config5:mysql5
```

Comment:
It's strange to add a mysql5 variant at this point, with the mysql5 port haven't been obsolete for so many years already. Variants for newer mysql software (mysql8, mysql57, mysql56, mysql55, mariadb-10.7, mariadb-10.6, mariadb-10.5) would be more reasonable, if this software supports that.

Reply:
Right, changed to support mysql8 and mysql57, reflecting gdal (minus mysql5).